### PR TITLE
Trigger endsession event after move

### PR DESCRIPTION
**Pull request by hutnikau**

### CODE REVIEW

File: views/js/runner/provider/qti.js [hunk 1]
```diff
@@ -318,6 +318,27 @@ define([
...
+             * @param {string} scope
+             * @param {integer} position
+             * @todo this kind of function is generic enough to be moved to a util/helper
+             * @returns {boolean}
+             */
+            function leaveSection(direction, scope, position)
+            {
+                var context = self.getTestContext();
+                var map     = self.getTestMap();
+                var section = map.parts[context.testPartId].sections[context.sectionId];
```
Comment (jsconan) [inline views/js/runner/provider/qti.js]: There is now a helper to do such actions:

```
mapHelper.getSection(map, context.sectionId);
```

File: views/js/runner/provider/qti.js [hunk 2]
```diff
@@ -318,6 +318,27 @@ define([
...
+                var nbItems = _.size(section.items);
+                var item    = section.items[context.itemIdentifier];
```
Comment (jsconan) [inline views/js/runner/provider/qti.js]: you can directly use the map helper to access to the item:

```
var item = mapHelper.getItem(map, context.itemIdentifier);
var sectionStats = mapHelper.getSectionStats(map, context.sectionId);
var nbItems = sectionStats && sectionStats.total;
```

**Comment by hutnikau** [comment]
@jsconan I updated the branch according with your comments and resolved conflicts in related branches.

File: manifest.php [hunk 3]
```diff
@@ -33,7 +33,7 @@
     'label' => 'QTI test model',
     'description' => 'TAO QTI test implementation',
     'license' => 'GPL-2.0',
-    'version' => '5.8.4',
+    'version' => '5.8.5',
     'author' => 'Open Assessment Technologies',
     'requires' => array(
         'taoTests' => '>=3.0.0',
```
Comment (jsconan) [inline manifest.php]: Isn't there a dependency to taoTests 3.3.1 ?

**Pull request merged**